go/merkletree/tree_hasher_test.go

```diff
@@ -0,0 +1,134 @@
+package merkletree
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"testing"
+)
+
+type leafTestVector struct {
+	inputLength int64
+	input       []byte
+	output      []byte
+}
+
+// Inputs and outputs are of fixed digest size.
+type nodeTestVector struct {
+	left   []byte
+	right  []byte
+	output []byte
+}
+
+type testVector struct {
+	emptyHash []byte
+	leaves    []leafTestVector
+	nodes     []nodeTestVector
+}
+
+const (
+	sha256EmptyHash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+)
+
+func dh(h string) []byte {
+	r, err := hex.DecodeString(h)
+	if err != nil {
+		panic(err)
+	}
+	return r
+}
+
+func getTestVector() testVector {
+	return testVector{
+		emptyHash: dh(sha256EmptyHash),
+		leaves: []leafTestVector{
+			{0, dh(""), dh("6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d")},
+			{1, dh("00"), dh("96a296d224f285c67bee93c30f8a309157f0daa35dc5b87e410b78630a09cfc7")},
+			{16, dh("101112131415161718191a1b1c1d1e1f"), dh("3bfb960453ebaebf33727da7a1f4db38acc051d381b6da20d6d4e88f0eabfd7a")},
+		},
+		nodes: []nodeTestVector{
+			{dh("000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f"),
```

Comment:
nit: these lines ought to be horizontally aligned. If the code formatter did this, I'm ok with leaving it this way though.